### PR TITLE
performed modifications after adding sqrt computation

### PR DESCRIPTION
**Pull request by ovatman**
Changes include:
- `UniswapV2Router02.addLiquidity()` and `UniswapV2Pair.mint()` functions are added (together with functions they use such as `math_sqrt()`)
- `testRouterAddLiquidity()` test added that checks for newly added functionality
- output from regression tests are updated

### CODE REVIEW

**Review by mariaKt — CHANGES_REQUESTED** [review]
I believe that this is failing due to compound assignment no longer being supported.

**Review by dwightguth — CHANGES_REQUESTED** [review]
Please also make sure you are applying the same modifications to the version of the contract in `test/regression/swaps.sol`. The two files differ slightly, but we want to keep them enough in sync that they are running the same computations. Also, that version is the only version that is currently being regression tested.

**Comment by dwightguth** [comment]
I will happily investigate the failures in the semantics if there are any, but now that we have a working version of the semantics on one version of the contract, changes to the contract need to only go through if they don't break the semantics.

**Comment by dwightguth** [comment]
Also, just fyi, auto formatting a file in the same PR that you make other modifications to that file is a bad idea; it makes it vastly more difficult to review the code.

**Comment by mariaKt** [comment]
I tried this version of Uniswap locally, and it does not run. It fails in` setUp` (https://github.com/Pi-Squared-Inc/solidity-demo-semantics/blob/ca576a1fbbd9a7285831d16a1c08140f510a6e2e/test/examples/swaps/UniswapV2Swap.sol#L673). Looking very briefly, it seems that is attempts to create a `UniswapV2Swap` using two different constructors.
https://github.com/Pi-Squared-Inc/solidity-demo-semantics/blob/ca576a1fbbd9a7285831d16a1c08140f510a6e2e/test/examples/swaps/UniswapV2Swap.sol#L677
and
https://github.com/Pi-Squared-Inc/solidity-demo-semantics/blob/ca576a1fbbd9a7285831d16a1c08140f510a6e2e/test/examples/swaps/UniswapV2Swap.sol#L689
Only the one with 3 parameters exists though.

**Comment by dwightguth** [comment]
Can you show me what the change to the contract and the resulting stuck configuration were?

**Comment by mariaKt** [comment]
@dwightguth if you meant me: I make no changes to the contract, and simply added the transaction
```
txn(1, 2, 0, 1724300000, testRouterAddLiquidity, )
``` after the others (so it is similar to the  UniswapTest.txn file in this PR). Here is the stuck configuration.

[conf.txt](https://github.com/user-attachments/files/16955291/conf.txt)

File: test/regression/swaps.txn [hunk 1]
```diff
@@ -1,2 +1,3 @@
 create(1, 0, 1724300000, UniswapV2SwapTest, ),
-txn(1, 2, 0, 1724300000, testSwapLoop, )
+txn(1, 2, 0, 1724300000, testSwapLoop, ),
+txn(1, 2, 0, 1724300000, testRouterAddLiquidity, )
```
Comment (dwightguth) [inline test/regression/swaps.txn:3]: This should not be here. This program tests just the swap function.

**Review by dwightguth — APPROVED** [review]
I'm approving so as to make you not stuck, but please address my comment prior to merging.

**Pull request merged**